### PR TITLE
Contributor guidelines: Document the project structure

### PR DESCRIPTION
**Pull request by buddhikac96**
-  projectStructure.md file added

- Code dive session video added.

### CODE REVIEW

File: doc/projectStructure.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    |   |_ AssemblyInfo.cs
+    |_ Main.cs
+    |_ NullableAttributes.cs
+    |_ winsw.csproj
+```
+
+#### :page_facing_up: Main.cs
+
+This file contains the entry point of the program. (Main method). This file includes the main flow of the program and has implemented the logics for command-line arguments.  
+ex : ```install, uninstall, start, stop, restart```
```
Comment (NextTurn) [inline doc/projectStructure.md:62]: What is "ex"?

File: doc/projectStructure.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    |_ WinSWCore.csproj
+    |_ WinSWException.cs
+    |_ WinSWSystem.cs
+    |_ Wmi.cs
+    |_ WmiSchema.cs
+```
+
+#### :page_facing_up: ServiceDescriptor.cs
+
+This contains the logics for extracting configurations from the XML file. ```ServiceDescriptor``` class get XML file as an argument. Currently, configurations are provided on demand.
```
Comment (NextTurn) [inline doc/projectStructure.md:88]: Use single backticks for code spans.

File: doc/projectStructure.md (hunk 1 above)
Comment (buddhikac96) [reply doc/projectStructure.md:62]: example :) its better to change into example.

File: doc/projectStructure.md (hunk 2 above)
Comment (buddhikac96) [reply doc/projectStructure.md:88]: Okay.. I'll do that. thank you so much for mention that :)

File: doc/projectStructure.md [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    |_ Properties
+    |   |_ AssemblyInfo.cs
+    |_ Main.cs
+    |_ NullableAttributes.cs
+    |_ winsw.csproj
+```
+
+#### :page_facing_up: Main.cs
+
+This file contains the entry point of the program. (Main method). This file includes the main flow of the program and has implemented the logics for command-line arguments such as `install`, `uninstall`, `start`, `stop`, `restart`.
```
Comment (oleg-nenashev) [inline doc/projectStructure.md:61]: Was changed in #535

File: doc/projectStructure.md [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+## :open_file_folder: src
+
+This contains the implementation of the WinSW.
+
+## :open_file_folder: Core
+
+## :notebook: ServiceWrapper
+
+This is the main executable project.
+```
```
Comment (oleg-nenashev) [inline doc/projectStructure.md:48]: I suggest removing it and summarizing the contents by text

File: doc/projectStructure.md [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+
+#### :page_facing_up: Main.cs
+
+This file contains the entry point of the program. (Main method). This file includes the main flow of the program and has implemented the logics for command-line arguments such as `install`, `uninstall`, `start`, `stop`, `restart`.
+
+## :notebook: WinSWCore
+
+```
+|_ WinSWCore
```
Comment (oleg-nenashev) [inline doc/projectStructure.md:66]: i suggest removing it and summarizing the content by text

File: doc/developer/projectStructure.md [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+## :open_file_folder: Core
+
+## :notebook: ServiceWrapper
+
+This is the main executable project. This contains the main.cs which is the entry point of the projet and contains some features for logging.
+
+### :page_facing_up: Main.cs
+
+This file contains the entry point of the program (Main method). This file includes the main flow of the program and has implemented the logics for following command line arguments.  
+ ```install, uninstall, start, stop, stopwait, restart, restart!, status, test, testwait, help, version```
```
Comment (oleg-nenashev) [inline doc/developer/projectStructure.md:34]: Rendering in GitHub editor went crazy there. Would it make sense to just reference documentation without providing a full list which will be obsolete soon anyway?

Comment (buddhikac96) [reply doc/developer/projectStructure.md:34]: Sure... Thank you for mention that !

Comment (buddhikac96) [reply doc/developer/projectStructure.md:34]: Sure. Thank you for mention that!

Comment (buddhikac96) [reply doc/developer/projectStructure.md]: Developer Guide is not working in `readme.md`. Shall I remove that as well.

**Comment by oleg-nenashev** [comment]
Thanks @buddhikac96 !

**Comment by NextTurn** [comment]
@oleg-nenashev This PR contains 7 commits and I think it would be better to squash and merge it.

**Pull request merged**